mycroft/skills/core.py

```diff
@@ -207,6 +207,67 @@ def real_decorator(func):
     return real_decorator
 
 
+#######################################################################
+# SkillGUI - Interface to the Graphical User Interface
+#######################################################################
+class SkillGUI(object):
+
+    def __init__(self, skill):
+        self.__dict = {}
+        self.page = None    # the active GUI page (e.g. QML template) to show
+        self.skill = skill
+
+    def __setitem__(self, key, value):
+        self.__dict[key] = value
+
+        if self.page:
+            # emit notification
+            data = self.__dict.copy()
+            data.update({'__from': self.skill.skill_id})
+            self.skill.bus.emit(Message("gui.value.set", data))
+
+    def __getitem__(self, key):
+        try:
+            return self.__dict[key]
+        except Exception:
+            return None
```

Comment:
I don't think it's a good idea to catch exceptions here. let the behavior be more like a normal dict and have a KeyError

Reply:
Fair enough